variables.tf

```diff
@@ -81,3 +87,79 @@ variable "existing_kms_instance_guid" {
   type        = string
   default     = null
 }
+
+########################################################################################################################
+# Cloud Directory variables
+########################################################################################################################
+
+variable "is_idp_cloud_directory_active" {
+  description = "Set this to true to set IDP Cloud Directory active."
+  type        = bool
+  default     = true
+}
+
+variable "is_mfa_active" {
+  description = "Set this to true to set MFA in IDP Cloud Directory active."
+  type        = bool
+  default     = true
+}
+
+variable "identity_confirm_access_mode" {
+  description = "Identity confirm access mode for Cloud Directory (CD). Allowed values are `FULL`, `RESTRICTIVE` and `OFF`."
+  type        = string
+  default     = "OFF"
+  validation {
+    condition     = contains(["FULL", "RESTRICTIVE", "OFF"], var.identity_confirm_access_mode)
+    error_message = "Allowed values for `identity_confirm_access_mode` are \"FULL\", \"RESTRICTIVE\" and \"OFF\"."
+  }
+}
+
+variable "identity_field" {
+  description = "Identity field for Cloud Directory (CD). Allowed values are `email` and `userName`."
+  type        = string
+  default     = "email"
+  validation {
+    condition     = contains(["email", "userName"], var.identity_field)
+    error_message = "Allowed values for `identity_field` are \"email\" and \"userName\"."
+  }
+}
+
+variable "reset_password_enabled" {
+  description = "Set this to true to enable password resets."
+  type        = bool
+  default     = false
+}
+
+variable "reset_password_notification_enabled" {
+  description = "Set this to true to enable password notifications."
+  type        = bool
+  default     = false
+}
+
+variable "signup_enabled" {
+  description = "Set this to true to allow users to signup."
+  type        = bool
+  default     = false
+}
+
+variable "self_service_enabled" {
+  description = "Set this to true to allow users to change password and edit user details."
+  type        = bool
+  default     = false
+}
+
+variable "welcome_enabled" {
+  description = "Set this to true to send welcome emails to the new users."
+  type        = bool
+  default     = false
+}
+
+########################################################################################################################
+# Users variables
+########################################################################################################################
+
+variable "users" {
+  description = "Number of users to add."
+  type        = list(string)
+  default     = ["user1234@example.com"]
```

Comment:
default should be empty list

Reply:
addresses in this PR as well as https://github.com/terraform-ibm-modules/terraform-ibm-appid/pull/145 for FScloud